File: sdk/js/examples/basic-usage.ts
/**
 * Basic SDK Usage Examples
 */

import { DataHubClient } from '../src/client';
import { ValidationError, NotFoundError } from '../src/errors';

// Initialize client
const client = new DataHubClient({
  baseUrl: process.env.DATAHUB_BASE_URL || 'https://api.hub.example.com/api/v1',
  apiToken: process.env.DATAHUB_API_TOKEN,
  enableLogging: true,
});

// Example 1: List assets
async function listAssets() {
  try {
    const assets = await client.get('/assets/', {
      params: {
        status: 'ACTIVE',
        limit: 20,
        offset: 0,
      },
    });
    console.log('Assets:', assets);
  } catch (error) {
    if (error instanceof NotFoundError) {
      console.error('Assets not found');
    } else {
      console.error('Error:', error);
    }
  }
}

// Example 2: Create asset
async function createAsset() {
  try {
    const asset = await client.post('/assets/', {
      key: 'my-asset',
      name: 'My Asset',
      description: 'Asset description',
      domain: 'marketing',
    });
    console.log('Created asset:', asset);
  } catch (error) {
    if (error instanceof ValidationError) {
      console.error('Validation errors:', error.details);
    } else {
      console.error('Error:', error);
    }
  }
}

// Example 3: Get asset by ID
async function getAsset(assetId: string) {
  try {
    const asset = await client.get(`/assets/${assetId}/`);
    console.log('Asset:', asset);
  } catch (error) {
    if (error instanceof NotFoundError) {
      console.error(`Asset ${assetId} not found`);
    } else {
      console.error('Error:', error);
    }
  }
}

// Example 4: Update asset
async function updateAsset(assetId: string) {
  try {
    const asset = await client.patch(`/assets/${assetId}/`, {
      description: 'Updated description',
    });
    console.log('Updated asset:', asset);
  } catch (error) {
    console.error('Error:', error);
  }
}

// Example 5: Delete asset
async function deleteAsset(assetId: string) {
  try {
    await client.delete(`/assets/${assetId}/`);
    console.log('Asset deleted');
  } catch (error) {
    console.error('Error:', error);
  }
}

// Run examples
async function main() {
  await listAssets();
  await createAsset();
  // await getAsset('asset-id');
  // await updateAsset('asset-id');
  // await deleteAsset('asset-id');
}

if (require.main === module) {
  main().catch(console.error);
}

File: sdk/python/examples/basic_usage.py
"""
Basic SDK Usage Examples
"""

import asyncio
import os

from datahub_interoperability import DataHubClient, DataHubClientConfig
from datahub_interoperability.errors import (
    NotFoundError,
    ValidationError,
)


async def list_assets(client: DataHubClient):
    """Example: List assets"""
    try:
        assets = await client.get(
            "/assets/",
            params={
                "status": "ACTIVE",
                "limit": 20,
                "offset": 0,
            },
        )
        print("Assets:", assets)
    except NotFoundError as e:
        print(f"Assets not found: {e.message}")
    except Exception as e:
        print(f"Error: {e}")


async def create_asset(client: DataHubClient):
    """Example: Create asset"""
    try:
        asset = await client.post(
            "/assets/",
            {
                "key": "my-asset",
                "name": "My Asset",
                "description": "Asset description",
                "domain": "marketing",
            },
        )
        print("Created asset:", asset)
    except ValidationError as e:
        print(f"Validation errors: {e.details}")
    except Exception as e:
        print(f"Error: {e}")


async def get_asset(client: DataHubClient, asset_id: str):
    """Example: Get asset by ID"""
    try:
        asset = await client.get(f"/assets/{asset_id}/")
        print("Asset:", asset)
    except NotFoundError as e:
        print(f"Asset {asset_id} not found: {e.message}")
    except Exception as e:
        print(f"Error: {e}")


async def update_asset(client: DataHubClient, asset_id: str):
    """Example: Update asset"""
    try:
        asset = await client.patch(
            f"/assets/{asset_id}/",
            {
                "description": "Updated description",
            },
        )
        print("Updated asset:", asset)
    except Exception as e:
        print(f"Error: {e}")


async def delete_asset(client: DataHubClient, asset_id: str):
    """Example: Delete asset"""
    try:
        await client.delete(f"/assets/{asset_id}/")
        print("Asset deleted")
    except Exception as e:
        print(f"Error: {e}")


async def main():
    """Main example"""
    # Initialize client
    config = DataHubClientConfig(
        base_url=os.getenv("DATAHUB_BASE_URL", "https://api.hub.example.com/api/v1"),
        api_token=os.getenv("DATAHUB_API_TOKEN"),
        enable_logging=True,
    )

    async with DataHubClient(config) as client:
        await list_assets(client)
        await create_asset(client)
        # await get_asset(client, "asset-id")
        # await update_asset(client, "asset-id")
        # await delete_asset(client, "asset-id")


if __name__ == "__main__":
    asyncio.run(main())
